drivers/include/saul.h

```diff
@@ -0,0 +1,157 @@
+ *
+ * Each device has further to expose a name and its type. This information can
+ * be used for automated searching and matching of devices (e.g. connect light
+ * sensor automatically with the color of an RGB LED...).
+ *
+ * The SAUL module enables further the automated initialization of preconfigured
+ * actuators/sensor via auto_init and the access to all available devices via
+ * one unified shell command.
+ *
+ * @todo        So far, the interface only supports simple read and set
+ *              operations. It probably needs to be extended to handling events,
+ *              thresholds, and so on.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Definition of the generic [S]ensor [A]ctuator [U]ber [L]ayer
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef SAUL_H
+#define SAUL_H
+
+#include <stdint.h>
+
+#include "phydat.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Definition of device classes
+ *
+ * This list contains a collections of available device classes. Each device
+ * must be part of one, but can be part of multiple of these classes. When
+ * belonging to more than one class, a device must however expose one driver
+ * for each class it belongs to, and it has to register each driver with a
+ * separate entry at the SAUL registry.
+ *
+ * Classes are identified by 8-bit unsigned integers.
+ *
+ * For searching and filtering purposes, the device classes are further split
+ * into two top-level classes: sensors and actuators. For identification, all
+ * actuator classes start with 0b01xxxxxx, all sensor classes start with
+ * 0b10xxxxxx.
+ *
+ * This list is not exhaustive, extend to your needs!
+ */
+enum {
+    SAUL_CLASS_UNDEF    = 0x00,     /**< device class undefined */
+    SAUL_ACT_ANY        = 0x40,     /**< any actuator - wildcard */
+    SAUL_ACT_LED_RGB    = 0x42,     /**< actuator: RGB LED */
+    SAUL_ACT_SERVO      = 0x43,     /**< actuator: servo motor */
+    SAUL_ACT_MOTOR      = 0x44,     /**< actuator: motor */
+    SAUL_ACT_SWITCH     = 0x45,     /**< actuator: simple on/off switch */
+    SAUL_ACT_DIMMER     = 0x46,     /**< actuator: dimmable switch */
+    SAUL_SENSE_ANY      = 0x80,     /**< any sensor - wildcart */
+    SAUL_SENSE_BTN      = 0x81,     /**< sensor: simple button */
+    SAUL_SENSE_TEMP     = 0x82,     /**< sensor: temperature */
+    SAUL_SENSE_HUM      = 0x83,     /**< sensor: humidity */
+    SAUL_SENSE_LIGHT    = 0x84,     /**< sensor: light */
+    SAUL_SENSE_ACCEL    = 0x85,     /**< sensor: accelerometer */
+    SAUL_SENSE_MAG      = 0x86,     /**< sensor: magnetometer */
+    SAUL_SENSE_GYRO     = 0x87,     /**< sensor: gyroscope */
+    SAUL_SENSE_COLOR    = 0x88,     /**< sensor: (light) color */
+    SAUL_SENSE_PRESS    = 0x89,     /**< sensor: pressure */
+    SAUL_SENSE_ANALOG   = 0x8a,     /**< sensor: raw analog value */
+    SAUL_CLASS_ANY      = 0xff,     /**< any device - wildcard */
```

Comment:
Remove the last comma to be pedantic- and C++ compliant.

see https://github.com/RIOT-OS/RIOT/pull/4358#discussion_r46080465